modules/margin-liquidity-pools/src/lib.rs

```diff
@@ -336,12 +356,30 @@ impl<T: Trait> MarginProtocolLiquidityPools<T::AccountId> for Module<T> {
 		Self::liquidity_pool_options(&pool_id, &pair).map(|pool| pool.ask_spread)
 	}
 
-	fn get_swap_rate(pool_id: Self::LiquidityPoolId, pair: Self::TradingPair) -> Fixed128 {
-		Self::swap_rate(pool_id, pair)
+	fn get_swap_rate(pool_id: Self::LiquidityPoolId, pair: Self::TradingPair, is_long: bool) -> Fixed128 {
+		if is_long {
+			let rate = Self::swap_rate(pair).unwrap_or_default().0;
+			if let Some(additional) = Self::additional_swap_rate(pool_id) {
+				rate.saturating_sub(additional)
```

Comment:
Why sub here instead of add?

Reply:
if default_swap_rate is `(long: -2, short: 2)` and additional_rate is 1, so the result is `(long: -3, short: 3)`

Reply:
short should be `2 - 1 = 1` right?

Reply:
```
long: -2 - 1 = -3
short: 2 + 1 = 3
```
the rate cost is trader pay to pool.

Reply:
> short: 2 + 1 = 3

but it is `saturating_sub` here?

Reply:
short in `else` code

Reply:
I see. Back to `(long: -2, short: 2)` => `(long: -3, short: 3)`, as traders pays pool if rate is positive, and pool pays trader if negative, this will lead to the situation that pool pays more and also gets more. But in this case, with additional rate 1, the pool pays less and get more I believe.